src/main/kotlin/com/jetbrains/snakecharm/lang/validation/SnakemakeSyntaxErrorAnnotator.kt

```diff
@@ -0,0 +1,65 @@
+package com.jetbrains.snakecharm.lang.validation
+
+import com.jetbrains.python.psi.PyKeywordArgument
+import com.jetbrains.snakecharm.SnakemakeBundle
+import com.jetbrains.snakecharm.lang.SnakemakeLanguageDialect
+import com.jetbrains.snakecharm.lang.psi.SMKRule
+import com.jetbrains.snakecharm.lang.psi.SMKRuleParameterListStatement
+import com.jetbrains.snakecharm.lang.psi.SMKRuleRunParameter
+
+object SnakemakeSyntaxErrorAnnotator : SnakemakeAnnotator() {
+    override fun visitSMKRuleParameterListStatement(st: SMKRuleParameterListStatement) {
+        if (!SnakemakeLanguageDialect.isInsideSmkFile(st)) {
+            return
+        }
+
+        val argumentNamesSet = HashSet<String>()
+        var encounteredKeywordArgument = false
+
+        st.argumentList?.arguments?.forEach {
+            // check argument name
+            val name = it.name
+            if (argumentNamesSet.contains(name)) {
```

Comment:
Could you refactor this method to make keyword argument processing more explicit? At the moment it isn't clear why we check `argumentNamesSet.contains(null)` in some cases, although it seems never can happen. Also `getName()` is method from `PsiElementBase` not clear how it is related to keyword arguments

E.g.
```
if (arg is PyKeywordArgument) {
  arg.keyword?.let { name ->
   ...
  }
  .. 
} else {
   // positional argument
  ...
}
```

Reply:
Please use `a in b` instead of `b.contains(a)`